utils/errors.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2017 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package utils
+
+type UsageError struct {
+	UserMsg string
+}
```

Comment:
Implement an Error interface on this one

Reply:
If you need to check if an `error` is a `UsageError` it's customary to implement something like:

```
func IsUsageErr(err error) {
    _, ok := err.(*UsageError)
    return ok
}
````